routes/auth.routes.js

```diff
@@ -0,0 +1,103 @@
+const { Router } = require("express");
+const bcrypt = require("bcrypt");
+const passport = require("passport");
+const router = Router();
+const saltRounds = 10;
+const User = require("../models/User.model");
+const { Mongoose } = require("mongoose");
+const mongoose = require("mongoose")
+
+router.get("/signup", (req,res,next) => res.render("auth/signup"));
+
+router.post("/signup", (req, res) => {
+    const { username, password } = req.body;
+
+    if (!username || !password){
+        if (password !== /^\S+@\S+\.\S+$/) {
+            res.render("auth/signup", {
+                validationError: "Please user a valid email adress.",
+                errorMessage:
+                "All fields are mandatory. Please provide your username, email and password.",
+            });
+        }
+        return;
+    }
+
+    const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+    if (!regex.test(password)) {
+    res
+      .status(500)
+      .render('auth/signup',
+        username,   
+        { errorMessage: '⛔️ Password needs to have at least 6 characters and must contain at least one number, one lowercase and one uppercase letter. ⛔️' });
```

Comment:
It's very nice that you are attempting to pass the ```username``` to the view as well. The only issue here is where you've placed the ```username```. This will throw off a server error, cause your views to crash and not show the ```errorMessage``` as you intend to do here.

You can change it by passing the username in the return object:
```suggestion
      .render('auth/signup', { username, errorMessage: '⛔️ Password needs to have at least 6 characters and must contain at least one number, one lowercase and one uppercase letter. ⛔️' });
```